src/arrayMethodJoin.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const mySeparator = separator === undefined ? ',' : separator;
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === '' || this[i] === undefined) {
+        result += '';
```

Comment:
let's not add empty string here
just add the element if it's not null or undefined